examples/fetch_ic/test/tests.ts

```diff
@@ -6,281 +6,221 @@ import * as dns from 'node:dns';
 dns.setDefaultResultOrder('ipv4first');
 
 import { getCanisterOrigin } from 'azle/dfx';
-import { Test } from 'azle/test';
+import { expect, it, please, Test, wait } from 'azle/test/jest';
 import puppeteer, { Browser, Page } from 'puppeteer';
 
-export function getTests(canisterName: string): Test[] {
+export function getTests(canisterName: string): Test {
     const origin = getCanisterOrigin(canisterName);
 
     let browser: Browser;
     let page: Page;
     let mainPage: Page;
 
-    return [
-        {
-            name: 'Prepare browser and pages',
-            prep: async () => {
-                browser = await puppeteer.launch({
-                    headless:
-                        process.env.GITHUB_RUN_ID === undefined ? false : true
-                });
-                page = await browser.newPage();
+    return () => {
+        please('prepare browser and pages', async () => {
+            browser = await puppeteer.launch({
+                headless: process.env.GITHUB_RUN_ID === undefined ? false : true
+            });
+            page = await browser.newPage();
 
-                page.on('console', (message) => {
-                    for (const arg of message.args()) {
-                        console.info(`Puppetteer log: ${arg}`);
-                    }
-                });
+            page.on('console', (message) => {
+                for (const arg of message.args()) {
+                    console.info(`Puppetteer log: ${arg}`);
+                }
+            });
 
-                await page.goto(origin);
+            await page.goto(origin);
 
-                await iiLogin(page);
+            await iiLogin(page);
 
-                const pages = await browser.pages();
+            const pages = await browser.pages();
 
-                mainPage = pages[1];
-            }
-        },
-        {
-            name: 'Waiting for identity to be set',
-            wait: 5_000
-        },
-        {
-            name: 'Headers Array',
-            test: async () => {
-                await mainPage.click('>>> #headersArrayButton');
-
-                await mainPage.waitForNetworkIdle();
-
-                const success = await page.evaluate(
-                    () => (window as any).headersArraySuccess
-                );
-
-                return {
-                    Ok: success
-                };
-            }
-        },
-        {
-            name: 'Headers Object',
-            test: async () => {
-                await mainPage.click('>>> #headersObjectButton');
+            mainPage = pages[1];
+        });
 
-                await mainPage.waitForNetworkIdle();
+        wait('for identity to be set', 5_000);
 
-                const success = await page.evaluate(
-                    () => (window as any).headersObjectSuccess
-                );
+        it('Headers Array', async () => {
```

Comment:
@lastmjs I'll need some help with the names for these test

Reply:
Yes we need to work on the names now based on the frontend tests